src/coreComponents/common/logger/ErrorHandling.cpp

```diff
@@ -0,0 +1,248 @@
+      index = stackLine.find( ':' );
+      m_sourceCallStack.push_back( stackLine.substr( index + 1 ) );
+    }
+  }
+
+  if( !m_isValidStackTrace )
+  {
+    m_sourceCallStack.push_back( std::string( g_callStackMessage ) );
+  }
+
+  return *this;
+}
+
+std::string ErrorLogger::toString( ErrorLogger::MsgType type )
+{
+  switch( type )
+  {
+    case ErrorLogger::MsgType::Error: return "Error";
+    case ErrorLogger::MsgType::Warning: return "Warning";
+    case ErrorLogger::MsgType::Exception: return "Exception";
+    default: return "Unknown";
+  }
+}
+
+void ErrorLogger::streamMultilineYamlAttribute( std::string_view msg, std::ofstream & yamlFile,
+                                                std::string_view indent )
+{
+  std::size_t i = 0;
+  // Loop that runs through the string_view named msg
+  while( i < msg.size() )
+  {
+    // Index of the next line break
+    std::size_t index = msg.find( "\n", i );
+    // If there is no line break, the entire string is taken
+    if( index == std::string_view::npos )
+    {
+      index = msg.size();
+    }
+    // Writes the current line to the YAML file with the desired indentation
+    std::string_view msgLine = msg.substr( i, index - i );
+    yamlFile << indent << msgLine << "\n";
+    // Move to the next line
+    i = index + 1;
+  }
+}
+
+void ErrorLogger::flushErrorMsg( ErrorLogger::ErrorMsg & errorMsg )
+{
+  std::ofstream yamlFile( std::string( m_filename ), std::ios::app );
+  if( yamlFile.is_open() )
+  {
+    // General errors info (type, rank on which the error occured)
+    yamlFile << "\n" << g_level1Start << "type: " << g_errorLogger.toString( errorMsg.m_type ) << "\n";
+    yamlFile << g_level1Next << "rank: ";
+    for( auto const & info: errorMsg.m_ranksInfo )
+    {
+      yamlFile << info;
+    }
+    yamlFile << "\n";
+    // Error message
+    yamlFile << g_level1Next << "message: >-\n";
+    streamMultilineYamlAttribute( errorMsg.m_msg, yamlFile, g_level2Next );
+    if( !errorMsg.m_contextsInfo.empty() )
+    {
+      // Additional informations about the context of the error and priority information of each context
+      yamlFile << g_level1Next << "contexts:\n";
+      for( ErrorContext const & ctxInfo : errorMsg.m_contextsInfo )
+      {
+        bool isFirst = true;
+        for( auto const & [key, value] : ctxInfo.m_attributes )
+        {
+          if( isFirst )
+          {
+            yamlFile << g_level3Start << ErrorContext::attributeToString( key ) << ": " << value << "\n";
+            isFirst = false;
+          }
+          else
+          {
+            yamlFile << g_level3Next << ErrorContext::attributeToString( key ) << ": " << value << "\n";
+          }
+        }
+        if( isFirst )
+        {
+          yamlFile << g_level3Start << "priority: " << ctxInfo.m_priority << "\n";
+          isFirst = false;
+        }
+        else
+        {
+          yamlFile << g_level3Next << "priority: " <<ctxInfo.m_priority << "\n";
+        }
```

Comment:
Would this shorter version work? (that would scale better if we add more hardcoded attributes)

```suggestion
        yamlFile << g_level3Start << "priority: " << ctxInfo.m_priority << "\n";
        for( auto const & [key, value] : ctxInfo.m_attributes )
        {
          yamlFile << g_level3Next << ErrorContext::attributeToString( key ) << ": " << value << "\n";
        }
```